src/bp/core/services/migration/index.ts

```diff
@@ -0,0 +1,105 @@
+import * as sdk from 'botpress/sdk'
+import { ConfigProvider } from 'core/config/config-loader'
+import { ModuleLoader } from 'core/module-loader'
+import { inject, injectable, tagged } from 'inversify'
+import _ from 'lodash'
+
+import Database from '../../database'
+import { TYPES } from '../../types'
+import { BotService } from '../bot-service'
+import { WorkspaceService } from '../workspace-service'
+
+import Migration11_9_2 from './migration_11_9_2'
+import Migration12_0_0 from './migration_12_0_0'
+
+export interface MigrationStep {
+  execute(dryRun?: boolean): Promise<sdk.MigrationStatus>
+}
+
+@injectable()
+export class MigrationService {
+  private migration12_0_0: Migration12_0_0
+  private migration11_9_2: Migration11_9_2
+
+  private _migrationPipeline: MigrationStep[] = []
+
+  constructor(
+    @inject(TYPES.Logger)
+    @tagged('name', 'MigrationService')
+    private logger: sdk.Logger,
+    @inject(TYPES.ConfigProvider) configProvider: ConfigProvider,
+    @inject(TYPES.BotService) botService: BotService,
+    @inject(TYPES.WorkspaceService) workspaceService: WorkspaceService,
+    @inject(TYPES.Database) private database: Database,
+    @inject(TYPES.ModuleLoader) private moduleLoader: ModuleLoader
+  ) {
+    this.migration12_0_0 = new Migration12_0_0(configProvider)
```

Comment:
Can't we just:

1) Scan files in this directoty
2) Sort them alphabetically
3) Execute them automatically in sequential order
4) By passing them an instance to the global IoC Container (Inversify), so they have free access to all services

Creating a new migration could be as simple as doing:

`yarn cmd create-migration`, which will generate a new file in this directory.  Done.
Same for the modules.  Inside the module, we do: `yarn create-migration`, which will create a new migration file from the module builder in the module's `migrations` folder.